File: backend/codehive/api/schemas/replay.py
"""Pydantic schemas for session replay endpoint."""

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class ReplayStep(BaseModel):
    """A single step in the session replay timeline."""

    index: int
    timestamp: datetime
    step_type: str
    data: dict[str, Any]


class ReplayResponse(BaseModel):
    """Paginated response for session replay."""

    session_id: uuid.UUID
    session_status: str
    total_steps: int
    steps: list[ReplayStep]

File: web/src/api/replay.ts
import { apiClient } from "./client";

export interface ReplayStep {
  index: number;
  timestamp: string;
  step_type: string;
  data: Record<string, unknown>;
}

export interface ReplayResponse {
  session_id: string;
  session_status: string;
  total_steps: number;
  steps: ReplayStep[];
}

export async function fetchReplay(
  sessionId: string,
  limit = 50,
  offset = 0,
): Promise<ReplayResponse> {
  const response = await apiClient.get(
    `/api/sessions/${sessionId}/replay?limit=${limit}&offset=${offset}`,
  );
  if (!response.ok) {
    throw new Error(`Failed to fetch replay: ${response.status}`);
  }
  return response.json() as Promise<ReplayResponse>;
}
